Base/Models/Public/Include/Models/UnifyedModel.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <map>
+#include <vector>
+#include <string>
+
+namespace Models
+{
+using FieldNames = std::initializer_list<std::string>;
+
+template <typename Enum>
+using FieldData = std::vector<std::tuple<Enum, size_t, std::string>>;
+
+template <typename TKey>
+class Comparator
+{
+public:
+    bool operator()(const TKey& first, const TKey& second) const noexcept
+    {
+        return static_cast<size_t>(first) < static_cast<size_t>(second);
+    }
+};
+
+template <typename TKey, typename TData = std::string, typename Comp = Comparator<TKey>>
+class Map : public std::map<TKey, TData, Comp>
```

Comment:
How are you going to implement this class later?

Reply:
Right now it's a main data type where I store information